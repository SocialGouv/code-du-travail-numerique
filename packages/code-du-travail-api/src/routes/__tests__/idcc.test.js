import router from "../idcc";

const request = require("supertest");
const Koa = require("koa");

const app = new Koa();
app.use(router.routes());

test("return idcc results for boulangerie", async () => {
  const response = await request(app.callback()).get(
    `/api/v1/idcc?q=boulangerie`
  );
  expect(response.status).toBe(200);
  expect(response.body.hits).toMatchSnapshot();
});

test("return idcc results for pati", async () => {
  const response = await request(app.callback()).get(`/api/v1/idcc?q=pati`);
  expect(response.status).toBe(200);
  expect(response.body.hits).toMatchSnapshot();
});

test("return idcc results in correct order for banque", async () => {
  const response = await request(app.callback()).get(`/api/v1/idcc?q=banque`);
  expect(response.status).toBe(200);
  expect(response.body.hits).toMatchSnapshot();
});

test("return idcc for number 0027", async () => {
  const response = await request(app.callback()).get(`/api/v1/idcc?q=0027`);
  expect(response.status).toBe(200);
  expect(response.body.hits).toMatchSnapshot();
});

test("return idcc for number 27", async () => {
  const response = await request(app.callback()).get(`/api/v1/idcc?q=27`);
  expect(response.status).toBe(200);
  expect(response.body.hits).toMatchSnapshot();
});
