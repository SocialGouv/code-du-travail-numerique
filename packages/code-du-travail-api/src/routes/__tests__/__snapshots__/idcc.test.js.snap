// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IDCC return idcc for number 27 1`] = `undefined`;

exports[`IDCC return idcc results for boulangerie 1`] = `undefined`;

exports[`IDCC return idcc results for pati 1`] = `undefined`;

exports[`IDCC return idcc results in correct order for banque 1`] = `undefined`;
